packages/@tailwindcss-upgrade/src/migrate-js-config.ts

```diff
@@ -121,10 +122,16 @@ async function migrateTheme(
 
     if (resetNamespaces.has(key[0]) && resetNamespaces.get(key[0]) === false) {
       resetNamespaces.set(key[0], true)
-      css += `  --${keyPathToCssProperty([key[0]])}-*: initial;\n`
+      let property = keyPathToCssProperty([key[0]])
+      if (property !== null) {
+        css += `  --${escape(property)}-*: initial;\n`
+      }
     }
 
-    css += `  --${keyPathToCssProperty(key)}: ${value};\n`
+    let property = keyPathToCssProperty(key)
+    if (property !== null) {
+      css += `  --${escape(property)}: ${value};\n`
+    }
```

Comment:
It might be wise to prepend the `--` before running escape here, because certain characters need to be escaped if they are the very first character but don't need to be escaped if they are not:

<img width="245" alt="image" src="https://github.com/user-attachments/assets/3c6f65f3-03be-4385-bd04-bfd18fae624f">